app/src/main/java/com/example/pokemonsinnowise/presentation/MainActivity.kt

```diff
@@ -0,0 +1,37 @@
+package com.example.pokemonsinnowise.presentation
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
+import com.example.pokemonsinnowise.databinding.ActivityMainBinding
+import com.example.pokemonsinnowise.presentation.di.PokemonDiFactory
+
+internal const val POKEMON = "pokemon"
+
+class MainActivity : AppCompatActivity() {
+
+    lateinit var viewModel: MainViewModel
```

Comment:
```suggestion
    private val viewModel: MainViewModel by viewModels()
```
Делегат `viewModels()` доступен в библиотеках `activity-ktx` и `fragment-ktx` для необходимых классов соответственно в рамках Jetpack и с параметрами конструктора для DI в случае с Hilt

Reply:
спасибо, поменял